doc: add release docs for bundle v0.9.0 (#20044)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/abi_compatibility.rst

   * - |NCS| versions
     - Compatible nRF54H20 SoC binaries version
   * - |NCS| v2.9.0-nRF54H20-1

>      - `nRF54H20 SoC Binaries v0.9.0`_, compatible with the nRF54H20 DK v0.9.0 and later revisions.

Reviewer: If you wanna add this here, you have to also add the corresponding link to the bundle in the links.txt file. If you search for nRF54H20 SoC Binaries v0. you will find the block where all the soc binaries links are located.

>    * - |NCS| v2.9.0-nRF54H20-1
>      - `nRF54H20 SoC Binaries v0.9.0`_, compatible with the nRF54H20 DK v0.9.0 and later revisions.

Reviewer: ```suggestion
   * - |NCS| v2.9.0-nRF54H20-1-rc2
     - `nRF54H20 SoC Binaries v0.9.1`_, compatible with the nRF54H20 DK v0.9.0 and later revisions.
```
plus the external link in the links.txt file

ABI compatibility ensures that the Secure Domain and System Controller firmware binaries do not need to be recompiled each time the application, radio binaries, or both are recompiled, as long as they are based on a compatible |NCS| version.
Additionally, maintaining ABI compatibility allows the nRF54H20 SoC binary components to work together without recompilation when updating to newer |NCS| versions.

nRF54H20 SoC Binaries v0.9.0 changelog
**************************************

The following sections provide detailed lists of changes by component.

Secure Domain Firmware (SDFW) v10.0.0
=====================================

Added
-----

> * SUIT in-place updatable components (IPUC) are now available to local domains over IPC through the SSF IPUC service.

Reviewer 2: This feature is apparently meant to be exposed through a higher level software component which is not ready yet, so it may not be worth mentioning here

* Purge protection can be enabled over ADAC.
* Clock control is enabled in SDFW.
* Global domain power request service is integrated in SDFW.
* PUF values from SDROM are cleared on boot.
* SUIT in-place updatable components (IPUC) are now available to local domains over IPC through the SSF IPUC service.
* GPIO DRIVECTRL for P6 and P7 on nRF54H20 is now corrected by SDFW on boot.
* Added support for TLS-1.3.
* Enable ED25519 pre-hashed.

Updated
-------

* Local domain reset will trigger a global reset.
  ``RESETINFO`` will contain both the global and local reset reason.
* All processors are booted regardless of whether they have firmware.

>   They are booted in halted mode if no firmware is present.

Reviewer 2: I think some of these changes would still be very noticeable to someone not having used the 0.8.0 version - do we want to remove them entirely?

Author: I agree, would an alternative be to not remove entries from old bundles? IMO all changes from all publicly available bundles should be listed here.

> * SDFW will now enter the suspend to RAM state when tracing is enabled.

Reviewer 2: The effect of this should be a noticeably reduced power consumption when tracing is enabled. Maybe that is more relevant to the reader here.

-------

* Local domain reset will trigger a global reset.

> * A local domain reset now triggers a global reset.  

Reviewer: Please remove line 74

  ``RESETINFO`` contains both the global and local reset reasons.

> * All processors now boot regardless of whether they have firmware.  

Reviewer: Please fix trailing whitespace

>   ``RESETINFO`` will contain both the global and local reset reason.

Reviewer: Fix also the previous line:

```suggestion
* A local domain reset now triggers a global reset.  
  ``RESETINFO`` contains both the global and local reset reasons.
```

> * All processors are booted regardless of whether they have firmware.
>   They are booted in halted mode if no firmware is present.

Reviewer: ```suggestion
* All processors now boot regardless of whether they have firmware.  
  If no firmware is present, they boot in halted mode.
```

> * Reduced power consumption from the secure domain when tracing is enabled.

Reviewer: ```suggestion
* Reduced power consumption from the Secure Domain when tracing is enabled.
```

> * Increased number of possible concurrent PSA operations to 8.

Reviewer: ```suggestion
* Increased the number of possible concurrent PSA operations to 8.
```

> * The ETR buffer location is now read from UICR.

Reviewer: ```suggestion
* The ETR buffer location is now read from the UICR.
```

>   It is required to configure the location when ETR tracing is enabled.

Reviewer: ```suggestion
  Enabling ETR tracing now requires configuring the location.
```

> * The SDFW will no longer immediately reset on a fatal error.

Reviewer: ```suggestion
* The SDFW no longer immediately resets on a fatal error.
```

Fixed
-----

> * SDFW no longer exits sleep state for a short duration after the boot has finished.

Reviewer: ```suggestion
* An issue where SDFW exited the sleep state for a short duration after boot completion.
```

* An issue where replies to ADAC SSF commands contained a large amount of additional zero values at the end of the message.

> * Permission checks for pointer members in SSF PSA crypto service requests are now correct.

Reviewer: ```suggestion
* An issue where permission checks for pointer members in the SSF PSA crypto service requests were incorrect.
```
